packages/orbit-components/src/Accordion/AccordionSection/components/SectionHeader.tsx

```diff
@@ -16,25 +17,82 @@ const AccordionSectionHeader = ({
   children,
   actions,
   expanded,
+  expandOnTileClick,
   onExpand,
   expandable,
   dataTest,
-}: Props) => (
-  <div
-    className={cx("p-600 flex items-center", expanded ? "min-h-[19px]" : "min-h-form-box-normal")}
-    data-test={dataTest && `${dataTest}Header`}
-  >
-    <div className="flex grow items-center">{children}</div>
-    {!expanded && expandable && (
-      <div className="ms-600 flex">
-        {actions || (
-          <Button onClick={onExpand} type="secondary">
-            Open
-          </Button>
+}: Props) => {
+  const isInteractive = expandOnTileClick && !expanded && expandable;
+
+  const handleClick = React.useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation();
+      onExpand?.();
+    },
+    [onExpand],
+  );
+
+  const handleKeyDown = React.useCallback(
+    (e: React.KeyboardEvent<HTMLButtonElement>) => {
+      if (e.key === "Enter" || e.key === " ") {
+        e.preventDefault();
+        onExpand?.();
+      }
+    },
+    [onExpand],
+  );
+
+  const handleButtonClick = React.useCallback(
+    (e: React.SyntheticEvent<HTMLButtonElement | HTMLAnchorElement>) => {
+      e.stopPropagation();
+      onExpand?.();
+    },
+    [onExpand],
+  );
+
+  const content = (
+    <>
+      <div className="flex grow items-center">{children}</div>
+      {!expanded && expandable && (
+        <div className="ms-600 flex">
+          {actions || (
+            <Button onClick={handleButtonClick} type="secondary">
+              Open
+            </Button>
+          )}
+        </div>
+      )}
+    </>
+  );
+
+  if (isInteractive) {
+    return (
+      <button
+        type="button"
+        className={cx(
+          "p-600 flex w-full items-center border-0 bg-transparent text-left",
+          expanded ? "min-h-[19px]" : "min-h-form-box-normal",
+          "hover:bg-cloud-light cursor-pointer",
         )}
-      </div>
-    )}
-  </div>
-);
+        onClick={handleClick}
+        onKeyDown={handleKeyDown}
+        data-test={dataTest && `${dataTest}Header`}
+        aria-expanded={expanded}
+      >
+        {content}
+      </button>
+    );
+  }
+
+  return (
+    <div
+      className={cx("p-600 flex items-center", expanded ? "min-h-[19px]" : "min-h-form-box-normal")}
+      data-test={dataTest && `${dataTest}Header`}
+      aria-expanded={expanded}
+    >
+      {content}
+    </div>
+  );
+};
```

Comment:
Can we merge these two to avoid code repetition? Use conditionals for the props.